src/client/providers/confirmation-provider.ts

```diff
@@ -0,0 +1,192 @@
+import { UserConfirmationProvider, ToolExecutionDetails, SettingsProvider, UserSettings } from '../types.js';
+import { logger } from '../../utils/logger.js';
+import * as readline from 'readline';
+import { FileSettingsProvider } from './file-settings-provider.js';
+import chalk from 'chalk';
+import boxen from 'boxen';
+
+/**
+ * Default implementation of UserConfirmationProvider
+ * Automatically approves tools in the allowedTools set,
+ * and prompts for confirmation for other tools using logger's styling
+ */
+export class CLIConfirmationProvider implements UserConfirmationProvider {
+    public allowedTools: Set<string>;
+    private rl: readline.Interface | null = null;
+    public settings: Promise<UserSettings>;
```

Comment:
instead of storing settings, store the `settingsProvider` and use that in this class to fetch settings - this let's the `settingsProvider` handle reading/writing all the settings

this will become handy later on if settings have more features later - we just need to add more methods into `settingsProvider`